requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c

```diff
@@ -0,0 +1,156 @@
+/* Body file for function Environment
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "client1.h"
+#include "routing.h"
+
+static asn1SccDataStoreCreateRequest create_request;
+static asn1SccDataStoreDeleteRequest delete_request;
+static asn1SccDataStoreRetrieveRequest retrieve_request;
+static asn1SccT_EventRetrieveLogMessage log_message;
+
+static int counter;
+static int subscribed;
+static int key;
+static asn1SccUShortInteger logIndex;
+
+static void print_log(asn1SccT_EventRetrieveLogMessage log_message)
+{
+    switch(log_message.u.log_item.operation.kind)
+    {
+        case DataStoreInternalLogItem_operation_item_created_PRESENT:
+            printf("client1: log data store item created\n");
+            break;
+
+        case DataStoreInternalLogItem_operation_item_deleted_PRESENT:
+            printf("client1: log data store item delete\n");
+            break;
+
+        case DataStoreInternalLogItem_operation_data_store_cleaned_PRESENT:
+            printf("client1: log data cleaned\n");
+            break;
+
+        default:
+            printf("unhandled log\n");
+    }
+}
+
+void client1_startup(void)
+{
+    counter = 0;
+    subscribed = 0;
+    key = 0;
+    logIndex = 0;
+}
+
+void client1_PI_Trigger( void )
+{
+    if(subscribed == 0)
+    {
+        asn1SccT_UInt32 event_id = event_id_datastore_notify;
+        asn1SccT_Boolean should_subscribe = 1;
+        client1_RI_subscribe_to_event(&event_id, &should_subscribe);
+        subscribed = 1;
+    }
+
+    if(counter < 12 && counter % 4 == 0)
+    {
+        create_request.behaviour = DataStoreCreateRequest_behaviour_free_existing;
+        create_request.item_value.kind = DataStoreValueType_coefficient_PRESENT;
+        create_request.item_value.u.coefficient = 2;
+        client1_RI_Create(&create_request);
+        ++counter;
+        ++key;
+    }
+    else if(counter < 12 && counter % 4 == 1)
+    {
+        client1_RI_RetrieveLogItem(&log_message, &logIndex);
+        print_log(log_message);
+        ++counter;
+    }
+    else if(counter < 12 && counter % 4 == 2)
+    {
+        delete_request.item_key = key;
+        client1_RI_Delete(&delete_request);
+        ++counter;
+    }
+    else if(counter < 12 && counter % 4 == 3)
+    {
+        client1_RI_RetrieveLogItem(&log_message, &logIndex);
```

Comment:
is it possible to add check for more than one entry?

Reply:
Done